raiden/messages.py

```diff
@@ -1416,6 +1416,217 @@ def from_dict(cls, data):
         return expired_lock
 
 
+class SignedBlindedBalanceProof(SignedMessage):
+    """Message sub-field `onchain_balance_proof` for `RequestMonitoring`.
+    """
+
+    def __init__(
+            self,
+            channel_identifier: typing.ChannelID,
+            token_network_address: typing.TokenNetworkAddress,
+            nonce: typing.Nonce,
+            additional_hash: typing.AdditionalHash,
+            chain_id: typing.ChainID,
+            signature: typing.Signature,
+            balance_hash: typing.BalanceHash,
+            non_closing_signature=b'',
```

Comment:
If would prefer to force the caller to pass either the `non_closing_signature` or the key to generate it, so that no instance of `SignedBlindedBalanceProof` can be unsigned. The same is true for other messages which require a signature.

Reply:
I agree with @konradkonrad , these classes are just PODs, the constructor is just setting the attributes and doing some value checking. Signing can be done by the `sign` function. (This may change in the near feature, since @andrevmatos wants to refactor these classes, so if we want to do interface changes it can go with andre's changes)